Specifications/QIR/Reserved-Operations.md

```diff
@@ -0,0 +1,41 @@
+# Reserved Operation Names
+
+As mentioned [here](Quantum-Runtime.md), we do reserve a set of quantum operation names.
+While targets are not required to implement any or all of these operations,
+if they define an operation with a reserved name that operation's effect is required
+to match the specified action.
+This ensures that quantum code that uses, say, a `CNOT` operation can rely on the
+effect of that operation being consistent across targets that define it.
+
+In this table, operation names should be considered case-insensitive.
+That is, if a target implements `RX` instead of `Rx`, its effect still must match that
+defined in the table.
+
+The matrices in this tables below take some time to load.
+Please be patient!
+
+| Operation Name | QIR Implementation | Description | Notes |
+|----------------|-----------------|-------------|-------|
+| H | [`__quantum__qis__h__body`](https://github.com/microsoft/qsharp-runtime/blob/5a379fd1c72ea19cb50d6acc501c6896894e96ac/src/Qir/Runtime/lib/QSharpCore/qsharp-core-qis.ll#L127) | The [Hadamard operation](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.intrinsic.h) | [Single-qubit operations](https://docs.microsoft.com/en-us/azure/quantum/concepts-the-qubit#single-qubit-operations), [Other Single-Qubit Cliffords](https://docs.microsoft.com/en-us/azure/quantum/user-guide/libraries/standard/prelude#other-single-qubit-cliffords) |
```

Comment:
I think rather than the link, I would put the explicit matrix here. Partly that's because the QIR specification shouldn't depend on the Q# documentation, and partly that's because the matrix is the precise definition of the gate.